agent/bench-scripts/pbench-trafficgen

```diff
@@ -1063,214 +1152,191 @@ function inner_most_loop() {
+			done
 			if [ $rc -eq 0 ]; then
-				echo "Iteration $iteration complete ($count of $total_iterations), with 1 pass and $failures failures"
-				log "Iteration $iteration complete ($count of $total_iterations), with 1 pass and $failures failures"
-			fi
-			if [[ "$postprocess_only" != "y" && -e "${tt_iteration_dir}" ]]; then
-				# Remove this iteration's "tool-trigger" directory.
-				rmdir ${tt_iteration_dir}
-				if [[ ${?} -ne 0 ]]; then
-					error_log "INTERNAL ERROR: failed to remove tool-trigger directory, '${tt_iteration_dir}'"
+				echo "$script_path/postprocess/process-iteration-samples \"$iteration_dir\" Mframes_sec \"$maxstddevpct\" \"$failures\" \"$max_failures\" \"$tar_nonref_data\" \"$keep_failed_tool_data\"" >"$iteration_dir/process-iteration-samples.cmd"
+				chmod +x "$iteration_dir/process-iteration-samples.cmd"
+				$iteration_dir/process-iteration-samples.cmd 2>&1 | tee $iteration_dir/process-iteration-samples.out
+				fail=$?
+				if [ $fail -ne 0 -a $fail -ne 1 ]; then
+					error_log "process-iteration-samples generated an unknown error, aborting"
 					exit 1
 				fi
+				if [ $fail -eq 1 ]; then
+					error_log "This test iteration failed"
+					((failures++))
+				fi
+				if [ $fail -eq 0 -o $failures -ge $max_failures ]; then
+					debug_log "Moving to the next iteration"
+					break
+				fi
+			else
+				error_log "Aborting benchmark"
+				break
+			fi
+		done # break out of this loop only if the $result_stddevpct is lower than $maxstddevpct
+		if [ $rc -eq 0 ]; then
+			echo_log "Iteration $iteration complete ($count of $total_iterations), with 1 pass and $failures failures"
+		fi
+		if [[ "$postprocess_only" != "y" && -e "${tt_iteration_dir}" ]]; then
+			# Remove this iteration's "tool-trigger" directory.
+			rmdir ${tt_iteration_dir}
+			if [[ ${?} -ne 0 ]]; then
+				error_log "INTERNAL ERROR: failed to remove tool-trigger directory, '${tt_iteration_dir}'"
+				exit 1
 			fi
-		else
-			echo "Skipping iteration $iteration ($count of $total_iterations)"
-			log "Skipping iteration $iteration ($count of $total_iterations)"
 		fi
-		let count=$count+1 # now we can move to the next iteration
 	done
 }
 
-count=1
+# Inner loop now pre-increments
+count=0
 rc=0
-if [ $traffic_generator == "trex-txrx-profile" -a "${traffic_profiles}" != "" ]; then
-	for traffic_profile in `echo $traffic_profiles | sed -e s/,/" "/g`; do
-		if [ $rc -ne 0 ]; then
+if [[ ${traffic_generator} == "trex-txrx-profile" && "${traffic_profiles}" != "" ]]; then
+	for traffic_profile in `echo ${traffic_profiles} | sed -e s/,/" "/g`; do
+		if [[ ${rc} -ne 0 ]]; then
 			break
 		fi
 
-		inner_most_loop "profile"
+		inner_most_loop "profile" "${traffic_profile}" "no-rate" "no-traffic-direction" "no-frame-size" "no-num-flow"
 	done
 else
-	for rate in `echo $rates | sed -e s/,/" "/g`; do
-		if [ $rc -ne 0 ]; then
+	for rate in `echo ${rates} | sed -e s/,/" "/g`; do
+		if [[ ${rc} -ne 0 ]]; then
 			break
 		fi
```

Comment:
I agree with moving it, perhaps when we migrate this code to the `main` branch?  I was trying not to make structural changes to the code where possible.  Most of the restructuring I took on was so that I could follow the logic of the code better, since it was pretty convoluted.